spec/02-integration/04-admin_api/15-off_spec.lua

```diff
@@ -536,6 +536,45 @@ describe("Admin API #off", function()
         assert.equals(snis.certificate.id, certificates.id)
       end)
 
+      it("snis in certificates without certificate", function()
+        local res = assert(client:send {
+          method = "POST",
+          path = "/config?flatten_errors=1",
+          body = {
+            _format_version = "1.1",
+            consumers = {
+              {
+                username = "x",
+                basicauth_credentials = {
+                  username = "x",
+                  password = "x",
+                }
+              }
+            },
+            certificates = {
+              {
+                cert = ssl_fixtures.cert,
+                id = "d83994d2-c24c-4315-b431-ee76b6611dcb",
+                key = ssl_fixtures.key,
+                snis = {
+                  {
+                    name = "foo.example",
+                    id = "1c6e83b7-c9ad-40ac-94e8-52f5ee7bde44",
+                  },
+                }
+              }
+            },
+          },
+          headers = {
+            ["Content-Type"] = "application/json"
+          }
+        })
+
+        local body = assert.response(res).has.status(400)
```

Comment:
The `400` error is due to DB-less by default does not support this kind of format.

It expects the `snis` not nested within the `certificates`.